### PR TITLE
Use reworked auth API

### PR DESCRIPTION
**Pull request by mr-kojo**
**Steps:**
- [x] E-mail verification
- [x] E-mail change
- [x] Password change
- [x] Password reset
- [x] Account deletion

### CODE REVIEW

**Comment by mr-kojo** [comment]
ToDos for password-change:

- [ ] Show an alert on success
- [ ] Clear fields on success 

ToDos for email-verification:

- [ ] Test verification if not logged in

**Comment by nicksellen** [comment]
@mr-kojo do you have time to continue this, or should I assign it to me? I think it's quite close now, looks good!

**Comment by mr-kojo** [comment]
Sorry for the delay, was busy with other things. Didn't forget it, though :) I'll continue to work in this this weekend.

File: src/components/Settings/ChangePassword.vue [hunk 1]
```diff
@@ -59,14 +59,13 @@ export default {
     }
   },
   methods: {
-    save () {
+    async save () {
```
Comment (mr-kojo) [inline src/components/Settings/ChangePassword.vue]: This new implementation of the `save` method no longer triggers an event for the parent component to call the backend. The reason is that this doesn't work in combination with `async`/`await`. 
As a result, the `changePassword` method of the `Settings` component (the parent component) is no longer used.
Do you consider this bad style? Is there a nicer way of clearing the password fields?

File: src/components/Settings/ChangePassword.vue [hunk 2]
```diff
@@ -59,8 +61,16 @@ export default {
   methods: {
     save () {
       const { oldPassword, newPassword } = this
-      this.$emit('save', { oldPassword, newPassword })
```
Comment (mr-kojo) [inline src/components/Settings/ChangePassword.vue]: This new implementation of the `save` method no longer triggers an event for the parent component to call the backend. The reason is that this doesn't work in combination with a Promise.
As a result, the `changePassword` method of the `Settings` component (the parent component) is no longer used.
Do you consider this bad style? Is there a nicer way of clearing the password fields?

Comment (nicksellen) [reply src/components/Settings/ChangePassword.vue]: It's not the style we've used so far, which would be to set some state (which might be status), but I don't feel like we've quite got all the common use cases covered - it seems very long winded for cases like this, and not really useful. It might be there is a reasonable similar approach that has that logic in a connect component, I don't think we've explored all the different options that those take.

Although maybe there is not time now to investigate further approaches, and that this is fine for now.

One thing though, it should use the async/await syntax which is equivelent to using the promises explicitly. For that you need to declare the whole `save()` function async (`async save()`) then just await on `this.saveUser()` (if there is an error it should throw, so should not need to check the result.

File: src/store/modules/auth.js [hunk 3]
```diff
@@ -93,7 +93,10 @@ export default {
 
       async changePassword ({ commit, state, dispatch }, data) {
         await auth.changePassword(data)
-        dispatch('logout')
+        dispatch('alerts/create', {
+          type: 'changePasswordSuccess',
+        }, { root: true })
+        return true
```
Comment (nicksellen) [inline src/store/modules/auth.js]: .... so can get rid of this `return true`

File: src/components/Settings/ChangePassword.vue (hunk 2 above)
Comment (nicksellen) [reply src/components/Settings/ChangePassword.vue]: The vuejs guy points out you can emit a callback in the event too... https://github.com/vuejs/vue/issues/5443#issuecomment-294147285

Comment (tiltec) [reply src/components/Settings/ChangePassword.vue]: We already pass actions as props in some other places (e.g. https://github.com/yunity/karrot-frontend/blob/f7377683baa1a4cbc1a328278f691f6e271c1499/src/components/Conversation/Conversation.vue#L120 ), so it seems fine to do the same here. Emitting a callback could be easier to understand though.

Comment (tiltec) [reply src/components/Settings/ChangePassword.vue]: Agree with switching `.then` to `await`, we almost always use `async`/`await` in our codebase (could refactor the rest at some point...)

Comment (mr-kojo) [reply src/components/Settings/ChangePassword.vue]: Aah, emitting a callback - that's clever. 
I implemented it this way now (6fd91c3).

File: src/pages/Settings.vue [hunk 4]
```diff
@@ -59,15 +59,14 @@ export default {
...
-        clearPasswordFields()
+    async changePassword (oldPassword, newPassword, done) {
+      await this.$store.dispatch('auth/changePassword', { oldPassword, newPassword })
+      if (!this.changePasswordStatus.hasValidationErrors) {
+        done()
       }
     },
     ...mapActions({
       saveUser: 'auth/save',
-      changePassword_: 'auth/changePassword',
```
Comment (mr-kojo) [inline src/pages/Settings.vue]: I replaced this line by the explicit call
`this.$store.dispatch('auth/changePassword', { oldPassword, newPassword })`
for the sake of better readability.

File: src/store/modules/users.js [hunk 5]
```diff
@@ -10,6 +10,7 @@ function initialState () {
     entries: {},
     idList: [],
     activeUserId: null,
+    resetPasswordSuccess: false,
```
Comment (mr-kojo) [inline src/store/modules/users.js]: Unrelated change

File: src/components/Login/PasswordReset.vue [hunk 6]
```diff
@@ -26,7 +26,7 @@
           class="error"
```
Comment (mr-kojo) [inline src/components/Login/PasswordReset.vue]: Unrelated change

File: src/components/Login/PasswordReset.vue [hunk 7]
```diff
@@ -26,7 +26,7 @@
...
         >
           <i class="fa fa-exclamation-triangle"/>
-          {{ $t('PASSWORD.RESET.CODE_ERROR') }}
+          {{ $t('GLOBAL.INVALID_LINK') }}
         </div>
```
Comment (mr-kojo) [inline src/components/Login/PasswordReset.vue]: Unrelated change

File: src/locales/locale-en.json [hunk 8]
```diff
@@ -272,8 +277,7 @@
     "RESET": {
       "TITLE": "Reset password",
       "OK": "Save",
-      "SUCCESS": "Your new password has been saved.",
-      "CODE_ERROR": "Sorry, the link you've opened is invalid."
+      "SUCCESS": "Your new password has been saved."
     },
```
Comment (mr-kojo) [inline src/locales/locale-en.json]: Unrelated change

**Comment by mr-kojo** [comment]
On my local machine, `yarn test` complains:
```code
[vuex] unknown getter: users/requestDeleteAccountStatus
```

What can I do about this?

**Comment by tiltec** [comment]
That happens if the store is not properly mocked in tests. I guess you added the `users/requestDeleteAccountStatus` getter to one of the components that is tested by Storyshots (`*.story.js` files). We usually don't include vuex in storyshots, therefore you need to either mock it or load the store implementation. Have a look at the other `.story.js` files, they have some store mocks and partial store implementations.

File: src/components/Settings/ChangePassword.vue [hunk 9]
```diff
@@ -1,25 +1,25 @@
 <template>
```
Comment (mr-kojo) [inline src/components/Settings/ChangePassword.vue]: Unrelated

**Comment by mr-kojo** [comment]
Should be ready to merge.
Some client-side form validation can be added later on (see #937).

**Comment by tiltec** [comment]
image

The sticky alert is a bit intense. Maybe it needs a different solution....

**Comment by tiltec** [comment]
I'd merge it without the changes above, it can be fixed later in `master`

**Comment by mr-kojo** [comment]
> The sticky alert is a bit intense. Maybe it needs a different solution....

That's right. I've already documented it here: #936
